src/test/java/com/example/bookstore/controller/CategoryControllerTest.java

```diff
@@ -0,0 +1,190 @@
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+    )
+    @DisplayName("Create a new category with valid request dto test")
+    void createBook_ValidRequestDto_Success() throws Exception {
+        CreateCategoryRequestDto requestDto = new CreateCategoryRequestDto()
+                .setName("Category")
+                .setDescription("Unreal category");
+
+        CategoryDto expected = new CategoryDto(
+                1L, requestDto.getName(), requestDto.getDescription()
+        );
+
+        String jsonRequest = objectMapper.writeValueAsString(requestDto);
+
+        MvcResult result = mockMvc.perform(
+                post("/api/categories")
+                        .content(jsonRequest)
+                        .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isCreated())
+                .andReturn();
+
+        CategoryDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(), CategoryDto.class
+        );
+
+        assertNotNull(actual);
+        assertNotNull(actual.id());
+        EqualsBuilder.reflectionEquals(expected, actual, "id");
+    }
+
+    @WithMockUser(username = "admin", authorities = {"ADMIN"})
+    @Test
+    @DisplayName("Create a new category with invalid request dto test")
+    void createBook_InvalidRequestDto_BadRequest() throws Exception {
+        CreateCategoryRequestDto requestDto = new CreateCategoryRequestDto()
+                .setName("")
+                .setDescription("");
+
+        String jsonRequest = objectMapper.writeValueAsString(requestDto);
+
+        mockMvc.perform(
+                post("/api/categories")
+                        .content(jsonRequest)
+                        .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isBadRequest())
+                .andReturn();
+    }
+
+    @WithMockUser(username = "admin", authorities = {"ADMIN"})
+    @Test
+    @DisplayName("Get All categories test")
+    void getAll_GivenCategories_Success() throws Exception {
+        List<CategoryDto> expected = new ArrayList<>();
+        expected.add(new CategoryDto(
+                1L, "Fantasy", "Fantasy books"));
+        expected.add(new CategoryDto(
+                2L, "Historical fiction", "Historical books"));
+        expected.add(new CategoryDto(
+                3L, "Mystery", "Mystery books"));
+
+        MvcResult result = mockMvc.perform(
+                get("/api/categories")
+                        .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+
+        CategoryDto[] actual = objectMapper.readValue(
+                result.getResponse().getContentAsByteArray(), CategoryDto[].class);
+
+        assertEquals(expected.size(), actual.length);
+        assertEquals(expected, Arrays.stream(actual).toList());
```

Comment:
Cannot say it's mandatory but your code formatting looks very strange for me
Please try following approach, maybe you also find it better
```suggestion
        List<CategoryDto> expected = new ArrayList<>();
        expected.add(new CategoryDto(1L, "Fantasy", "Fantasy books"));
        expected.add(new CategoryDto(2L, "Historical fiction", "Historical books"));
        expected.add(new CategoryDto(3L, "Mystery", "Mystery books"));

        MvcResult result = mockMvc
                .perform(get("/api/categories").contentType(MediaType.APPLICATION_JSON))
                .andExpect(status().isOk())
                .andReturn();

        CategoryDto[] actual = objectMapper.readValue(
                result.getResponse().getContentAsByteArray(), CategoryDto[].class);

        assertEquals(expected.size(), actual.length);
        assertEquals(expected, Arrays.stream(actual).toList());
```